File: lib/java/src/main/java/io/github/telepact/internal/types/TFieldDeclaration.java
//|
//|  Copyright The Telepact Authors
//|  SPDX-License-Identifier: Apache-2.0
//|

package io.github.telepact.internal.types;

public class TFieldDeclaration {
    public final String fieldName;
    public final TTypeDeclaration typeDeclaration;
    public final boolean optional;

    public TFieldDeclaration(
            String fieldName,
            TTypeDeclaration typeDeclaration,
            boolean optional) {
        this.fieldName = fieldName;
        this.typeDeclaration = typeDeclaration;
        this.optional = optional;
    }
}

File: lib/py/telepact/internal/types/TFieldDeclaration.py
#|
#|  Copyright The Telepact Authors
#|  SPDX-License-Identifier: Apache-2.0
#|

from typing import Optional, TYPE_CHECKING

if TYPE_CHECKING:
    from .TTypeDeclaration import TTypeDeclaration


class TFieldDeclaration:
    def __init__(
            self,
            field_name: str,
            type_declaration: 'TTypeDeclaration',
            optional: bool
    ) -> None:
        self.field_name = field_name
        self.type_declaration = type_declaration
        self.optional = optional
